### PR TITLE
Fluentd-Cloudwatch Changes

### LINKED ISSUE #2839: [incubator/fluentd-cloudwatch] should support RBAC
**Is this a request for help?**: NO
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): FEATURE REQUEST
**Version of Helm and Kubernetes**: Helm: v2.6.1  k8s: 1.7
**Which chart**: [incubator/fluentd-cloudwatch]
**What happened**: Fluentd-cloudwatch chart should support RBAC, with some minimal clusterrole. A deployment will not work out of the box with a RBAC enabled cluster.
**What you expected to happen**: Sane defaults for a RBAC enabled cluster.
**How to reproduce it** (as minimally and precisely as possible): Deploy the chart.
**Anything else we need to know**: Nope

### PR DESCRIPTION
**Pull request by jwhitcraft**
- Add support for awsRole vai kube2iam role
- Add rbac create support (Fixes #2839)
- Standardize helpers with other charts

### CODE REVIEW

**Comment by foxish** [comment]
cc/ @kenden @icereval @jmcarp as maintainers of this chart.

File: incubator/fluentd-cloudwatch/templates/_helpers.tpl [hunk 1]
```diff
@@ -2,15 +2,16 @@
 {{/*
 Expand the name of the chart.
 */}}
-{{- define "fluentd-cloudwatch.name" -}}
-{{- default .Chart.Name .Values.nameOverride | trunc 24 -}}
+{{- define "name" -}}
```
Comment (icereval) [inline incubator/fluentd-cloudwatch/templates/_helpers.tpl]: You'll want to keep the chart name prefix, more info @ https://github.com/kubernetes/charts/issues/1785

File: incubator/fluentd-cloudwatch/templates/_helpers.tpl [hunk 2]
```diff
@@ -2,15 +2,16 @@
...
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
 {{/*
 Create a default fully qualified app name.
 We truncate at 24 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
 */}}
-{{- define "fluentd-cloudwatch.fullname" -}}
+{{- define "fullname" -}}
```
Comment (icereval) [inline incubator/fluentd-cloudwatch/templates/_helpers.tpl]: Keep the chart name prefix, more info @ https://github.com/kubernetes/charts/issues/1785

File: incubator/fluentd-cloudwatch/README.md [hunk 3]
```diff
@@ -53,6 +56,7 @@ The following tables lists the configurable parameters of the Fluentd Cloudwatch
 | `hostNetwork`                   | Host network                               | `false`                                                    |
 | `annotations` (removed for now) | Annotations                                | `nil`                                                      |
 | `awsRegion`                     | AWS Cloudwatch region                      | `us-east-1`                                                |
+| `awsRole`                       | AWS IAM Role To Use                        | `nil`                                                      |
```
Comment (icereval) [inline incubator/fluentd-cloudwatch/README.md:59]: please add documentation for `rbac.create` and `rbac.serviceAccountName`

**Review by icereval — CHANGES_REQUESTED** [review]
A couple small changes, thanks for the help!

File: incubator/fluentd-cloudwatch/README.md (hunk 3 above)
Comment (jwhitcraft) [reply incubator/fluentd-cloudwatch/README.md:59]: @icereval thanks for the pointers, this the rbac.create and rbac.serviceAccountName were added.

**Comment by icereval** [comment]
LGTM

**Pull request merged**